core/src/wallet/ethereum/ERC20/ERC20LikeAccount.cpp

```diff
@@ -72,15 +73,72 @@ namespace ledger {
         std::shared_ptr<api::BigInt> ERC20LikeAccount::getBalance() {
             auto result = BigInt::ZERO;
             auto operations = getOperations();
+
             for (auto &op : operations) {
-                if (op->getOperationType() == api::OperationType::RECEIVE) {
-                    result = result + BigInt(op->getValue()->toString(10));
-                } else if (op->getOperationType() == api::OperationType::SEND){
-                    result = result - BigInt(op->getValue()->toString(10));
+                result = accumulateBalanceWithOperation(result, *op);
+            }
+
+            return std::make_shared<api::BigIntImpl>(result);
+        }
+
+        api::ERC20LikeBalanceHistory ERC20LikeAccount::getBalanceHistoryFor(
+            const std::chrono::system_clock::time_point& startDate,
+            const std::chrono::system_clock::time_point& endDate,
+            api::TimePeriod period
+        ) {
+            // guard against bad arguments
+            if (startDate >= endDate) {
+                throw make_exception(
+                    api::ErrorCode::INVALID_DATE_FORMAT,
```

Comment:
INVALID_ARGUMENT more suitable, but we didn't distinguish them anyway...

Reply:
I just do it the way it’s done everywhere else (i.e. Bitcoin and Ethereum).

Reply:
ok